crates/dekaf/src/connector.rs

```diff
@@ -70,8 +71,22 @@ pub async fn unary_materialize(
 ) -> anyhow::Result<materialize::Response> {
     use proto_flow::materialize::response::validated;
     if let Some(mut validate) = request.validate {
-        serde_json::de::from_str::<DekafConfig>(&validate.config_json)
-            .context("validating endpoint config")?;
+        match materialization_spec::ConnectorType::try_from(validate.connector_type)? {
```

Comment:
Just as an fyi, you could also use

```rust
let ConnectorType::Dekaf = ConnectorType::try_from(validate.connector_type)? else {
  bail!("...");
};
```

Reply:
But then I wouldn't have the parsed `proto_flow::flow::materialization_spec::ConnectorType`, which can give me a human-readable`as_str_name()` instead of just `proto_flow::materialize::request::Validate:: connector_type: i32`, which is much less useful in an error